engine/gui/widgets.py

```diff
@@ -106,10 +108,86 @@ def create(
                 None,
             )
 
-        return GUISpec(
+        gui_spec = GUISpec(
             assets=[Asset(**a) for a in spec.get("assets", [])],
             buttons=buttons,
             images=[Image(**b) for b in spec.get("images", [])],
             cancel_action=cancel_action,
             initial_selected_button=selected_button,
         )
+
+        cls.validate(gui_spec)
+        return gui_spec
+
+    def validate(self) -> None:
```

Comment:
Class methods should be decorated with `@classmethod`, and use `cls` instead of `self`. This isn't so much a style thing as much as a Python language thing: https://docs.python.org/3/library/functions.html#classmethod

In converting this to a class method you'd want to accept an argument:
```python
@classmethod
def validate(cls, gui_spec) -> None:
```

Alternatively you could keep this as an instance method, and call it on the object instead of the class:
```python
gui_spec.validate()
```

I'd say the second approach would make more sense, but up to you.